vault/router_test.go

```diff
@@ -477,3 +505,50 @@ func TestPathsToRadix(t *testing.T) {
 		t.Fatalf("bad: %v (sub/bar)", raw)
 	}
 }
+
+func TestParseUnauthenticatedPaths(t *testing.T) {
+	// inputs
+	paths := []string{
+		"foo",
+		"foo/*",
+		"sub/bar*",
+	}
+	wildcardPaths := []string{
+		"end/+",
+		"+/begin/*",
+		"middle/+/bar*",
+	}
+	allPaths := append(paths, wildcardPaths...)
+
+	p, err := parseUnauthenticatedPaths(allPaths)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// outputs
+	wildcardPathsEntry := map[string]bool{
+		"end/+":         true,
+		"+/begin/*":     true,
+		"middle/+/bar*": true,
+	}
+	expected := &loginPathsEntry{
+		paths:         pathsToRadix(paths),
+		wildcardPaths: wildcardPathsEntry,
+	}
+
+	if !reflect.DeepEqual(expected, p) {
+		t.Fatalf("expected: %#v\n actual: %#v\n", expected, p)
+	}
+}
+
+func TestParseUnauthenticatedPaths_Error(t *testing.T) {
+	// inputs
+	invalidPaths := []string{
+		"/foo/+*",
```

Comment:
Might be outside the original scope of this PR, but I don't think we're enforcing `*` to be at the end of a path even though we should.

Reply:
I think it is a good idea to add more validation here. I added some more validation and test cases. Please let me know what you think.